src/server.c

```diff
@@ -2737,7 +2737,8 @@ void initServer(void) {
     evictionPoolAlloc(); /* Initialize the LRU keys pool. */
     server.pubsub_channels = dictCreate(&keylistDictType);
     server.pubsub_patterns = dictCreate(&keylistDictType);
-    server.pubsubshard_channels = dictCreate(&keylistDictType);
+    server.pubsubshard_channels = dictCreateMultiple(&keylistDictType, slotCount);
```

Comment:
Nice follow up.

The only drawback that I can think of is the increased fixed memory overhead, especially for nodes that don't use sharded pubsub at all. Can we initialize this array of dicts lazily? We can let it be NULL until the first shard channel is created, then we allocate the array.

And for nodes with only very few shard channels, we can create only the dicts which are non-empty and let the others be NULL until we need them. It is the same discussion as we had for keys, but shard channels are much less used than keys and there is less code accessing the shard channels, so it is not as complex to created them lazily as it would be to do the same for keys.